File: Android/2016-04-29-ButterKnifeExample/app/src/main/java/tech/thdev/butter_knife_example/adapter/model/PhotoDataModel.java
package tech.thdev.butter_knife_example.adapter.model;

import tech.thdev.butter_knife_example.network.domain.Photo;

/**
 * Created by Tae-hwan on 5/3/16.
 */
public interface PhotoDataModel {

    void add(Photo photo);

    void add(Photo photo, boolean isNotify);

    void remove(int position);

    Photo getPhotoItem(int position);

    int getSize();
}

File: kotlin-example-01/src/main/kotlin/tech/thdev/kotlin_example_01/view/adapter/model/PhotoDataModel.kt
package tech.thdev.kotlin_example_01.view.adapter.model

import tech.thdev.kotlin_example_01.data.Photo

/**
 * Created by Tae-hwan on 7/22/16.
 */
interface PhotoDataModel {

    fun addItem(item: Photo)

    fun clean()
}